File: PHP/Interface/get_ratings.php
<?
/* this script will handle the variables passed from the insert_customer.html file */

/* declare some relevant variables */

PRINT "<html>";
PRINT "<head><title>Get Ratings</title></head>";
PRINT "<body>";

include 'connection.php';


$cname = @$_POST["cname"];
if($cname != ""){$cname .= ",";}
$citycity = @$_POST["citycity"];
if($citycity != ""){$citycity .= ",";}
$mtitle = @$_POST["mtitle"];
if($mtitle != ""){$mtitle .= ",";}
$albname = @$_POST["albname"];
if($albname != ""){$albname .= ",";}
$artname = @$_POST["artname"];
if($artname != ""){$artname .= ",";}
$yyear = @$_POST["yyear"];
if($yyear != ""){$yyear .= ",";}
$ggenre = @$_POST["ggenre"];
if($ggenre != ""){$ggenre .= ",";}

/* MySQL table created to store the data */
$table0 = "city";
$table1 = "customer";
$table2 = "music";
$table3 = "year";
$table4 = "album";
$table5 = "artist";
$table6 = "rating";
$table7 = "attribute";
$table8 = "genre";

PRINT "<center>";

  $query = "SELECT $cname $citycity $mtitle $albname $artname $yyear $ggenre r.rating
            FROM $table0 city, $table1 c, $table2 m, $table3 y, $table4 alb, $table5 art, $table6 r, $table7 a, $table8 g   WHERE c.cid = r.cid and c.cityid = city.cityid and r.mid = m.mid and m.aid = a.aid and a.gid = g.gid and a.yid = y.yid and a.artid = art.artid and a.albid = alb.albid";
  
$result = MYSQL_QUERY($query);

/* How many rows in the result? */
if(!$result){
  $number = 0;
}else{
  $number = MYSQL_NUMROWS($result);
}

if ($number != 0)
{
 $i = 0;
 PRINT "Rating(s): $number<BR>";
 PRINT "<table border = 1>";
 PRINT "<tr>";
 if($cname != ""){PRINT "<td>Name</td>";}
 if($citycity != ""){PRINT "<td>City</td>";}
 if($mtitle != ""){PRINT "<td>Title</td>";}
 if($albname != ""){PRINT "<td>Album</td>";}
 if($artname != ""){PRINT "<td>Artist</td>";}
 if($yyear != ""){PRINT "<td>Year</td>";}
 if($ggenre != ""){PRINT "<td>Genre</td>";}
 PRINT "<td>Rating</td>";
 PRINT "</tr>";
 while ($i < $number){
    PRINT "<tr>";
    
    if($cname != ""){
      $name = mysql_result($result, $i, "c.name");
      PRINT "<td>$name</td>";}
    if($citycity != ""){
      $city = mysql_result($result, $i, "city.city");
      PRINT "<td>$city</td>";}
    if($mtitle != ""){
      $title = mysql_result($result, $i, "m.title");
      PRINT "<td>$title</td>";}
    if($albname != ""){
      $album = mysql_result($result, $i, "alb.name");
      PRINT "<td>$album</td>";}
    if($artname != ""){
      $artist = mysql_result($result, $i, "art.name");
      PRINT "<td>$artist</td>";}
    if($yyear != ""){
      $year = mysql_result($result, $i, "y.year");
     PRINT "<td>$year</td>";}
    if($ggenre != ""){
      $genre = mysql_result($result, $i, "g.genre");
      PRINT "<td>$genre</td>";}
      
    $rating = mysql_result($result, $i, "r.rating");
    PRINT "<td>$rating</td>";

    PRINT "</tr>";
    $i++;
 }
 PRINT "</table>";
}
else
{
  PRINT "<strong>No Ratings in Database $table6</strong><br>";
}

PRINT "</center>";
PRINT "<hr style=\"margin-left:5em; margin-right:5em\">";
PRINT "<center>";
PRINT "<a href = \"get_ratings.html\">Back</a>";
PRINT "</center>";
PRINT "</body>";
PRINT "</html>";

/* Close the database connection */
MYSQL_CLOSE();

?>

File: Servlets/Interface/db2/src/java/dbase2/musicdb/get_ratings.java
package dbase2.musicdb;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletResponse;

import java.io.IOException;
import java.io.PrintWriter;

import java.sql.Connection;
import java.sql.SQLException;
import java.sql.Statement;
import java.sql.DriverManager;
import java.sql.ResultSet;

public class get_ratings extends HttpServlet{
  
  public void doGet(HttpServletRequest request, HttpServletResponse response)
    throws ServletException, IOException{
    response.setContentType("text/html");
    PrintWriter writer = response.getWriter();
    writer.write("<html>\n");
    writer.write("<head><title>Get Ratings</title></head>\n");
    writer.write("<body>\n");
    writer.write("<center>\n");

    String cname = request.getParameter("cname");
    if((cname != null) || (cname.length() != 0))
      cname += ",";

    String citycity = request.getParameter("citycity");
    if((citycity != null) || (citycity.length() != 0))
      citycity += ",";
    
    String mtitle = request.getParameter("mtitle");
    if((mtitle != null) || (mtitle.length() != 0))
      mtitle += ",";

    String albname = request.getParameter("albname");
    if((albname != null) || (albname.length() != 0))
      albname += ",";
    
    String artname = request.getParameter("artname");
    if((artname != null) || (artname.length() != 0))
      artname += ",";

    String yyear = request.getParameter("yyear");
    if((yyear != null) || (yyear.length() != 0))
      yyear += ",";

    String ggenre = request.getParameter("ggenre");
    if((ggenre != null) || (ggenre.length() != 0))
      ggenre += ",";
    
    try{
      Class.forName("com.mysql.jdbc.Driver");
    }catch(ClassNotFoundException exc){
      exc.printStackTrace();
      printMsg("No JDBC driver found!", writer, request);
      return;
    }
    try{
      Connection connection = DriverManager.getConnection("jdbc:mysql://localhost/db2", "student", "student");
      Statement statement = connection.createStatement(); //referential integrity
      ResultSet result = null;
      
       result = statement.executeQuery("SELECT "+cname +citycity +mtitle +albname +artname +yyear +ggenre+" r.rating FROM city city, customer c, music m, year y, album alb, artist art, rating r, attribute a, genre g WHERE c.cid = r.cid and c.cityid = city.cityid and r.mid = m.mid and m.aid = a.aid and a.gid = g.gid and a.yid = y.yid and a.artid = art.artid and a.albid = alb.albid");
     
    if(result != null){
      writer.write("Rating(s): <BR>\n");
      writer.write("<table border = 1>\n");
      writer.write("<tr>\n");
      if(cname != "")
        writer.write("<td>Name</td>\n");
      if(citycity != "")
        writer.write("<td>City</td>\n");
      if(mtitle != "")
        writer.write("<td>Title</td>\n");
      if(albname != "")
        writer.write("<td>Album</td>\n");
      if(artname != "")
        writer.write("<td>Artist</td>\n");
      if(yyear != "")
        writer.write("<td>Year</td>\n");
      if(ggenre != "")
        writer.write("<td>Genre</td>\n");
      writer.write("<td>Rating</td>\n");
      writer.write("</tr>\n");
      
      while(result.next()){
        String name = result.getString("c.name");
        String city = result.getString("city.city");
        String title = result.getString("m.title");
        String album = result.getString("alb.name");
        String artist = result.getString("art.name");
        String year = ((new Integer(result.getInt("y.year"))).toString());
        String genre = result.getString("g.genre");
        String rating = ((new Integer(result.getInt("r.rating"))).toString());

        writer.write("<tr>\n");
        if(cname != "")
          writer.write("<td>"+name+"</td>\n");
        if(citycity != "")
          writer.write("<td>"+city+"</td>\n");
        if(mtitle != "")
          writer.write("<td>"+title+"</td>\n");
        if(albname != "")
          writer.write("<td>"+album+"</td>\n");
        if(artname != "")
          writer.write("<td>"+artist+"</td>\n");
        if(yyear != "")
          writer.write("<td>"+year+"</td>\n");
        if(ggenre != "")
          writer.write("<td>"+genre+"</td>\n");
        writer.write("<td>"+rating+"</td>\n");
        writer.write("</tr>\n");
      }
        writer.write("</table>\n");
    }

    printMsg("", writer, request);
    }catch(SQLException exc){
      exc.printStackTrace();
      printMsg("Database error!", writer, request);
    }
  }

  private void printMsg(String msg, PrintWriter writer, HttpServletRequest request){
    writer.write("<h1>" + msg + "</h1>\n");
    writer.write("</center>\n");
    writer.write("<hr style=\"margin-left:5em; margin-right:5em\">\n");
    writer.write("<center>\n");
    writer.write("<a href = \"get_ratings.html\">Back</a>\n");
    writer.write("</center>\n");
    writer.write("</body>\n");
    writer.write("</html>\n");
  }
}
